### PR TITLE
Copy the Extension and KubernetesConfig structs from aks-engine.

### PR DESCRIPTION
**Pull request by yizhang4321**
This removes our dependency on those two structs from aks-engine code base. Our end goal is to remove all aks-engine dependencies.

### CODE REVIEW

File: pkg/agent/datamodel/types.go [hunk 1]
```diff
@@ -359,39 +359,129 @@ type LinuxProfile struct {
...
+	AzureCNIURLWindows                string                `json:"azureCNIURLWindows,omitempty"`
+	KeyVaultSku                       string                `json:"keyVaultSku,omitempty"`
+	MaximumLoadBalancerRuleCount      int                   `json:"maximumLoadBalancerRuleCount,omitempty"`
+	ProxyMode                         api.KubeProxyMode     `json:"kubeProxyMode,omitempty"`
+	PrivateAzureRegistryServer        string                `json:"privateAzureRegistryServer,omitempty"`
+	OutboundRuleIdleTimeoutInMinutes  int32                 `json:"outboundRuleIdleTimeoutInMinutes,omitempty"`
+}
+
 // MasterProfile represents the definition of the master cluster
 type MasterProfile struct {
```
Comment (xuto2) [inline pkg/agent/datamodel/types.go:453]: we can remove this later

File: pkg/agent/datamodel/types.go [hunk 2]
```diff
@@ -403,10 +493,10 @@ type MasterProfile struct {
...
 // OrchestratorProfile contains Orchestrator properties
 type OrchestratorProfile struct {
-	OrchestratorType    string                `json:"orchestratorType"`
-	OrchestratorVersion string                `json:"orchestratorVersion"`
-	KubernetesConfig    *api.KubernetesConfig `json:"kubernetesConfig,omitempty"`
-	DcosConfig          *api.DcosConfig       `json:"dcosConfig,omitempty"`
+	OrchestratorType    string            `json:"orchestratorType"`
+	OrchestratorVersion string            `json:"orchestratorVersion"`
+	KubernetesConfig    *KubernetesConfig `json:"kubernetesConfig,omitempty"`
+	DcosConfig          *api.DcosConfig   `json:"dcosConfig,omitempty"`
```
Comment (xuto2) [inline pkg/agent/datamodel/types.go:499]: we can remove this later

File: pkg/agent/datamodel/types.go (hunk 1 above)
Comment (yizhang4321) [reply pkg/agent/datamodel/types.go:453]: Yes, I can remove the code not needed? Do you mean we can remove "PrivateAzureRegistryServer" and "OutboundRuleIdleTimeoutInMinutes" fields or the "MasterProfile" struct?

File: pkg/agent/datamodel/types.go (hunk 2 above)
Comment (yizhang4321) [reply pkg/agent/datamodel/types.go:499]: Do you mean all 4 fields (OrchestratorType, OrchestratorVersion, KubernetesConfig, and DcosConfig) or just the DcosConfig field?

**Pull request merged**